src/DataValues/DecimalMath.php

```diff
@@ -252,115 +252,62 @@ private function roundDigits( $value, $significantDigits ) {
 			return '+0';
 		}
 
+		$len = strlen( $value );
+
+		// keeping all digits means no rounding
+		if ( $significantDigits >= ( $len -1 ) ) {
```

Comment:
$significantDigits is documented in roundToDigit(): $significantDigits The number of digits to retain, counting the decimal point, but not counting the leading sign.

Counting the decimal point makes it easier to use $significantDigits as a string offset. But it's still bad, we should change it. But I didn't want to mess with it in this change. 

The -1 is needed because the first character is the sign. It's "+1234", not "1234".

So: the sign is guaranteed to be there, and $significantDigits includes the decimal point (even though that is silly). That's why the offset calculation works.